.github/workflows/test.yml

```diff
@@ -169,15 +169,25 @@ jobs:
       - run: ruby -v
 
   testLatestRubygemsVersion:
-    name: "Test rubygems: latest upgrades the default RubyGems version"
+    name: "Ruby ${{ matrix.ruby }}: latest upgrades RubyGems"
     runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+        - { ruby: '3.2', rg_vers: '3.5.3'  }
+        - { ruby: '3.0', rg_vers: '3.5.3'  }
+        - { ruby: '2.7', rg_vers: '3.4.22' }
+        - { ruby: '2.6', rg_vers: '3.4.22' }
+        - { ruby: '2.5', rg_vers: '3.3.27' }
+        - { ruby: '2.3', rg_vers: '3.3.27' }
     steps:
     - uses: actions/checkout@v4
     - uses: ./
       with:
-        ruby-version: '2.6'
+        ruby-version: ${{ matrix.ruby }}
         rubygems: latest
-    - run: ruby -e "exit(Gem.rubygems_version > Gem::Version.new('3.0.3'))"
+    - run: ruby -e "puts Gem::VERSION; exit(Gem.rubygems_version >= Gem::Version.new('${{ matrix.rg_vers }}'))"
```

Comment:
Could you swap single and double quotes? It's safer to use single quotes outside for Ruby code (was already an issue before but since you're touching this line let's do it)